auth/http.go

```diff
@@ -1527,6 +1529,46 @@ func httpHandlerHasActiveDirectory(central *ImqsCentral, w http.ResponseWriter,
 	}
 }
 
+// httpHanderGetDynamicPermissions returns all of the dynamic/client specific permissions
+// as set in the imqsauth config file.
+func httpHandlerGetDynamicPermissions(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
+	response := []byte("{}")
+	var err error
+	if central.Config.Permissions != nil {
+		response, err = json.Marshal(central.Config.Permissions)
+		if err != nil {
+			authaus.HttpSendTxt(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+	}
+	w.Header().Add("content-type", "application/json")
+	httpSendResponse(w, response)
+}
+
+// httpHanderGetDynamicPermissionsList returns a stripped down map of the dynamic permissions mapping
+// the permissions ID to its name. This map is intended to by used by services to find the ID of a
+// permissions provided the name of the permission is known.
+func httpHanderGetDynamicPermissionsList(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
```

Comment:
Why do we need this second API?

Reply:
If developers want to use a permission in a service they would like use the permission name in the service instead of the ID. This allows the service to pull all of the dynamic permission names with their corresponding ID's for a quick lookup of the permissions ID.